packages/react-utils/src/helpers/componentUtils.tsx

```diff
@@ -35,13 +35,17 @@ export const PrivateComponent = (props: ComponentProps) => {
     opensrpBaseURL: configs.opensrpBaseURL,
   };
   const extraData = useSelector((state) => getExtraData(state));
+  const authenticated = useSelector((state) => isAuthenticated(state));
   const { roles } = extraData;
   const { activeRoles } = props;
-  return activeRoles && roles && isAuthorized(roles, activeRoles) ? (
-    <ConnectedPrivateRoute {...CPRProps} />
-  ) : (
-    <UnauthorizedPage title={lang.FORBIDDEN_PAGE_STATUS} />
-  );
+  if (authenticated) {
+    if (activeRoles && roles && isAuthorized(roles, activeRoles)) {
+      return <ConnectedPrivateRoute {...CPRProps} />;
+    } else {
+      return <UnauthorizedPage title={lang.FORBIDDEN_PAGE_STATUS} />;
+    }
+  }
+  return <ConnectedPrivateRoute {...CPRProps} />;
```

Comment:
Redirect to login instead here

Reply:
> Redirect to login instead here

Is there provision for authenticated but not authorised?  This would mean if unauthenticated it redirects to login and if unauthorised to unauthorised

Reply:
Yes if unauthenticated it should redirect to login, if authenticated but not authorized should redirect to the Unauthorized page

Reply:
We can as well leave it that way because `ConnectedPrivateRoute` redirects to login if user is unauthenticated